runtime/src/main/jni/Runtime.cpp

```diff
@@ -301,9 +309,39 @@ jint Runtime::GenerateNewObjectId(JNIEnv *env, jobject obj)
 	return objectId;
 }
 
-void Runtime::AdjustAmountOfExternalAllocatedMemoryNative(JNIEnv *env, jobject obj, jlong usedMemory)
+void Runtime::AdjustAmountOfExternalAllocatedMemory() {
+	int64_t usedMemory = m_env.CallLongMethod(m_runtime, GET_USED_MEMORY_METHOD_ID);
+	int64_t changeInBytes = usedMemory - m_lastUsedMemory;
+	int64_t externalMemory = 0;
+
+	if (changeInBytes != 0) {
+		externalMemory = m_isolate->AdjustAmountOfExternalAllocatedMemory(changeInBytes);
+	}
+
+	DEBUG_WRITE("usedMemory=%" PRId64 " changeInBytes=%" PRId64 " externalMemory=%" PRId64, usedMemory, changeInBytes, externalMemory);
+
+	m_lastUsedMemory = usedMemory;
+}
+
+bool Runtime::NotifyGC(JNIEnv *env, jobject obj)
 {
-	m_isolate->AdjustAmountOfExternalAllocatedMemory(usedMemory);
+	bool success = __sync_bool_compare_and_swap(&m_runGC, false, true);
+	return success;
+}
+
+bool Runtime::TryCallGC() {
```

Comment:
Also this info: 

LowMemoryNotification forces up to 7 compacting GCs. While IdleNotificationDeadline (the IdleNotification version is deprecated and will go away soon) tries to do some garbage collection work that fits between now and the deadline. If you feed V8 enough idle time via this API, you should see almost no GC activity during script execution.